angular-ngrx-scss/src/app/shared/components/repo-controls/repo-controls.component.html

```diff
@@ -7,39 +7,37 @@
       placeholder="Find a repository..."
       [formControl]="searchInput"
     />
-    <button class="control-dropdown" role="button">
-      <span class="control-dropdown__name">Type</span>
-      <span
-        class="control-dropdown__caret"
-        appOcticon="chevron-down"
-        size="14"
-      ></span>
-    </button>
-    <button class="control-dropdown" role="button">
-      <span class="control-dropdown__name">Language</span>
-      <span
-        class="control-dropdown__caret"
-        appOcticon="chevron-down"
-        size="14"
-      ></span>
-    </button>
-    <button class="control-dropdown" role="button">
-      <span class="control-dropdown__name">Sort</span>
-      <span
-        class="control-dropdown__caret"
-        appOcticon="chevron-down"
-        size="14"
-      ></span>
-    </button>
+    <app-repos-filter-dropdown
+      name="Type"
+      description="Select type"
+      [items]="typeFilters"
+      [current]="selectFilterByType$ | async"
+      (setFilter)="handleTypeClick($event)"
+    ></app-repos-filter-dropdown>
+    <app-repos-filter-dropdown
+      name="Language"
+      description="Select type"
+      [items]="typeFilters"
+      (setFilter)="handleTypeClick($event)"
+    ></app-repos-filter-dropdown>
+    <app-repos-filter-dropdown
+      name="Sort"
+      description="Select type"
+      [items]="typeFilters"
+      (setFilter)="handleTypeClick($event)"
+    ></app-repos-filter-dropdown>
   </div>
   <ng-container *ngIf="hasActiveSortAndFilters$ | async">
     <div class="divider"></div>
     <div class="results">
       <p class="results__text">
-        <ng-container *ngIf="selectReposCount$ | async as count">
-          <span class="bold">{{ count }}</span>
+        <ng-container *ngIf="{ data: selectReposCount$ | async } as count">
```

Comment:
[curious]: how does this ```*ngIf="{ data: selectReposCount$ | async } ``` work please?

Reply:
or more like, why are we wrapping it in an object first?

Reply:
I'm wrapping in an object because if I rely on the ngIf as it is the `0 results` will not show because 0 is a falsy value.

Reply:
Ah, this is interesting lol. In that case, what if we did away with the ng-container entirely?

for example, I am assuming the conditional check if just to hide it till we can get a first value off the observable?
I think (do not know for sure but maybe we can check) we can simply have something like this in that case

```
 <span class="bold">{{ selectReposCount$ | async  }}</span>
```

Directly, I think the number would only get rendered after the first emit, which should achieve the same thing? Let me know what you think about this?

I'm seeing similar direct use in the docs https://angular.io/api/common/AsyncPipe